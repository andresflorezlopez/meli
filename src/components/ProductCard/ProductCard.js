import React from "react";
import PropTypes from "prop-types";
import Typography from "@components/Typography/Typography";
import {
  StyledImage,
  StyledProductCard,
  StyledProductShipping,
  StyledProductLocation,
  StyledProductDescription,
  StyledProductInformation,
  StyledProductImageContainer
} from "./ProductCard.styled.js";

const ProductCard = ({
  id,
  image,
  price,
  title,
  location,
  freeShipping,
  onClick
}) => {
  const renderPrice = () =>
    freeShipping ? (
      <>
        ${price}
        <StyledProductShipping />
      </>
    ) : (
      `$ ${price}`
    );

  return (
    <StyledProductCard onClick={() => onClick(id)}>
      <StyledProductImageContainer>
        <StyledImage src={image} />
      </StyledProductImageContainer>
      <StyledProductInformation>
        <StyledProductDescription>
          <Typography type="h3">{renderPrice()}</Typography>
          <Typography type="medium">{title}</Typography>
        </StyledProductDescription>
        <StyledProductLocation>
          <Typography type="small" textAlign="center">
            {location}
          </Typography>
        </StyledProductLocation>
      </StyledProductInformation>
    </StyledProductCard>
  );
};

ProductCard.propTypes = {
  image: PropTypes.string,
  freeShipping: PropTypes.bool,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired
};

ProductCard.defaultProps = {
  freeShipping: false,
  background: "transparent"
};
export default ProductCard;
