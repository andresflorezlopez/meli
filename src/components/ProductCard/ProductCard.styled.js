import styled from "styled-components";
import shippingIcon from "@assets/images/shipping.png";

export const StyledProductCard = styled.div`
  display: flex;
  flex: 2;
  cursor: pointer;
  flex-direction: row;
  margin: ${({ theme }) => theme.spacing(2)};
  margin-bottom: 0;
  padding-bottom: ${({ theme }) => theme.spacing(2)};
  border-bottom: 1px solid ${({ theme }) => theme.colors.background};
`;

export const StyledProductImageContainer = styled.div`
  width: 180px;
  height: 180px;
  overflow: hidden;
  margin-right: ${({ theme }) => theme.spacing(2)};
`;

export const StyledImage = styled.img`
  width: 180px;
  border-radius: ${({ theme }) => theme.borderRadius.normal};
  height: auto;
`;

export const StyledProductInformation = styled.div`
  display: flex;
  flex: 8;
  flex-direction: row;
`;

export const StyledProductDescription = styled.div`
  flex: 4;
  padding: ${({ theme }) => theme.spacing(2)} 0;
  h3 {
    margin-bottom: ${({ theme }) => theme.spacing(4)};
    ${({ theme }) => theme.breakpoints.sm} {
      margin-bottom: ${({ theme }) => theme.spacing(1)};
    }
  }
`;

export const StyledProductShipping = styled.div`
  border-radius: ${({ theme }) => theme.borderRadius.total};
  margin-left: ${({ theme }) => theme.spacing(1)};
  margin-top: ${({ theme }) => theme.spacing(0.5)};
  background: url(${shippingIcon});
  background-repeat: no-repeat;
  background-position: center;
  display: inline-block;
  height: 20px;
  width: 20px;
`;

export const StyledProductLocation = styled.div`
  padding: ${({ theme }) => theme.spacing(4)} 0;
  flex: 3;
  ${({ theme }) => theme.breakpoints.sm} {
    display: none;
  }
`;
