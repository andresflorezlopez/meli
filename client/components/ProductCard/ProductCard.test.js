import React from "react";
import ProductCard from "./ProductCard";
import { mount } from "enzyme";

import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("ProductCard", () => {
  test("should render with freeShipping", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <ProductCard
          id="123"
          image="https://picsum.photos/id/180/180/180"
          price={1980}
          title="Macbook pro retina 2020 casi nuevo completo unico fabuloso!"
          location="Mendoza"
          freeShipping={true}
          onClick={id => console.log(id)}
        />
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render without freeShipping", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <ProductCard
          id="1223"
          image="https://picsum.photos/id/180/180/180"
          price={1980}
          title="Macbook pro retina 2020 casi nuevo completo unico fabuloso!"
          location="Mendoza"
          freeShipping={false}
          onClick={id => console.log(id)}
        />
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should call onClick event", () => {
    const spy = jest.fn();
    const wrapper = mount(
      <ThemeProvider theme={LightTheme}>
        <ProductCard
          id="1f23"
          image="https://picsum.photos/id/180/180/180"
          price={1980}
          title="Macbook pro retina 2020 casi nuevo completo unico fabuloso!"
          location="Mendoza"
          freeShipping={false}
          onClick={id => spy(id)}
        />
      </ThemeProvider>
    );
    wrapper
      .find("div")
      .first()
      .simulate("click");
    expect(spy).toHaveBeenCalled();
  });
});
