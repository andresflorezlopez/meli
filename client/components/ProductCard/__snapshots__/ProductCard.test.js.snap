// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render with freeShipping 1`] = `
<div
  className="y2k85v-0 jxQjHS"
  onClick={[Function]}
>
  <div
    className="y2k85v-1 kIXagQ"
  >
    <img
      className="y2k85v-2 fIdIsj"
      src="https://picsum.photos/id/180/180/180"
    />
  </div>
  <div
    className="y2k85v-3 bdLeSw"
  >
    <div
      className="y2k85v-4 fUuhkr"
    >
      <h3
        className="obo2vd-2 dwHcWW"
        color="normal"
        fontWeight="normal"
        type="h3"
      >
        $
        1.980
        <div
          className="y2k85v-5 dvNhyE"
        />
      </h3>
      <p
        className="obo2vd-3 hxdhNI"
        color="normal"
        fontWeight="normal"
        type="medium"
      >
        Macbook pro retina 2020 casi nuevo completo unico fabuloso!
      </p>
    </div>
    <div
      className="y2k85v-6 bkcQQl"
    >
      <p
        className="obo2vd-6 dYbffT"
        color="normal"
        fontWeight="normal"
        type="small"
      >
        Mendoza
      </p>
    </div>
  </div>
</div>
`;

exports[`ProductCard should render without freeShipping 1`] = `
<div
  className="y2k85v-0 jxQjHS"
  onClick={[Function]}
>
  <div
    className="y2k85v-1 kIXagQ"
  >
    <img
      className="y2k85v-2 fIdIsj"
      src="https://picsum.photos/id/180/180/180"
    />
  </div>
  <div
    className="y2k85v-3 bdLeSw"
  >
    <div
      className="y2k85v-4 fUuhkr"
    >
      <h3
        className="obo2vd-2 dwHcWW"
        color="normal"
        fontWeight="normal"
        type="h3"
      >
        $ 1.980
      </h3>
      <p
        className="obo2vd-3 hxdhNI"
        color="normal"
        fontWeight="normal"
        type="medium"
      >
        Macbook pro retina 2020 casi nuevo completo unico fabuloso!
      </p>
    </div>
    <div
      className="y2k85v-6 bkcQQl"
    >
      <p
        className="obo2vd-6 dYbffT"
        color="normal"
        fontWeight="normal"
        type="small"
      >
        Mendoza
      </p>
    </div>
  </div>
</div>
`;
